textures/metro_control_panel_001_pegboard.py

```diff
@@ -25,10 +32,95 @@ def _remove_empty(pegs: Pegs) -> Pegs:
 
 
 def _swap_pegs(pegs: Pegs, a: str, b: str) -> Pegs:
+    """Trade places of two symbols"""
     assert len(a) == 1 and len(b) == 1
 
     placeholder = "_"
     # noinspection PyTypeChecker
     return tuple(
         tuple(s.replace(a, placeholder).replace(b, a).replace(placeholder, b) for s in panel) for panel in pegs
     )
+
+
+def markers_to_bools(peg_str: str) -> Tuple[bool]:
+    """
+    The positions of the markers represented as booleans
+
+    :param peg_str: String of peg symbols
+    :return: List of booleans, true where each marker (M) was
+
+    >>> markers_to_bools("PME")
+    [False, True, False]
+    """
+    return tuple(peg == "M" for peg in peg_str)
+
+
+def pegs_to_bools(peg_str: str) -> Tuple[bool]:
+    """
+    The positions of the markers represented as booleans
+
+    :param peg_str: String of peg symbols
+    :return: List of booleans, true where each peg (P) was
+
+    >>> markers_to_bools("PME")
+    [True, False, False]
+    """
+    return tuple(peg == "E" for peg in peg_str)
+
+
+def invert_bools(bits: Iterable[bool]) -> Tuple[bool]:
+    return tuple(not bit for bit in bits)
+
+
+def bools_to_nor_mask(masking_bools: Collection[bool]) -> NorMask:
+    """
+    Creates an elementwise function that can called with two lists of booleans
+    of the same shape.
+    How the returned nor mask function is meant to be used:
+    (A, B, and C are lists; * is the nor mask function)
+    A * B = C
+    For C[n] the value A[n] is given where masking_bools[n] is False
+    and A[n] NOR B[n] is given where masking_bools[n] is True
+
+    :param masking_bools: Bools that determine if the function should NOR or not
+    :return: Elementwise operator
+
+    >>> a = [False, True, False, True]
+    >>> b = [False, False, True, True]
+    >>> nor_mask_false = bools_to_nor_mask([False, False, False, False])
+    >>> nor_mask_false(a, b)
+    [False, True, False, True]
+    >>> nor_mask_true = bools_to_nor_mask([True, True, True, True])
+    >>> nor_mask_true(a, b)
+    [True, False, False, False]
+    """
+
+    def nor(a: bool, b: bool) -> bool:
+        return not (a or b)
+
+    # noinspection PyUnusedLocal
```

Comment:
it _is_ used, so why the suppression?

Reply:
That's very strange. I can live with it, but I'd prefer it wouldn't be necessary without convoluting the code.